### PR TITLE
Provide JSX code linting, add pre-commit ESLint hook, add Travis CI

### LINKED ISSUE #7: Improve code linting
There're a few issues connected to the linting topic:

- JSX files aren't validated (ESLint doesn't parse the JSX files)
- no pre-commit hook - we can provide it using `husky` and `lint-staged` packages
- existing errors in JSX files should be fixed
- we can add linting to the CI (Travis)

### PR DESCRIPTION
**Pull request by ma2ciek**
Internal: Provided JSX code linting, added `pre-commit` ESLint hook, added Travis CI. Closes #7.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -6,15 +6,25 @@
...
+  env: {
+    mocha: false
+  },
+  settings: {
+    react: {
+      version: '16.3'
+    }
+  },
+  rules: {
+    'react/prop-types': 'off'
```
Comment (elszczepano) [inline .eslintrc.js]: Maybe it's a good idea to add `propTypes` to ensure if a component received props with exact type?

File: backend/package.json [hunk 2]
```diff
@@ -1,17 +1,15 @@
 {
-  "name": "HangHub-backend",
+  "name": "hang-hub-backend",
```
Comment (pjasiun) [inline backend/package.json]: I think it should be `hanghub-backend`. We do not use HangHub as 2 separate words. It is like "CKEditor". We use "ckeditor" when we need to use all lowwer case, not "c-k-editor".

File: frontend/package.json [hunk 3]
```diff
@@ -1,14 +1,13 @@
...
-  "description": "HangHub is a productivity tool that allows you to see users who are working on the same GitHub issue as you. It is available as a browser extension for Chrome and Firefox.",
-  "main": "src/index.js",
-  "scripts": {
-    "build": "webpack --mode=production",
-    "lint": "eslint ./src"
-  },
+  "name": "hang-hub-frontend",
+  "version": "0.0.1",
+  "private": true,
+  "description": "",
```
Comment (pjasiun) [inline frontend/package.json]: Why did you remove the description?

File: package.json [hunk 4]
```diff
@@ -0,0 +1,32 @@
+{
+  "name": "hang-hub",
```
Comment (pjasiun) [inline package.json]:
```suggestion
  "name": "hanghub",
```

File: .eslintrc.js [hunk 5]
```diff
@@ -6,15 +6,25 @@
...
+    ecmaFeatures: {
+      jsx: true
     }
+  },
+  env: {
+    mocha: false
+  },
+  settings: {
+    react: {
+      version: '16.3'
```
Comment (elszczepano) [inline .eslintrc.js:24]: Can we update the React version there to `16.8.6` (newest)?

File: frontend/package.json [hunk 6]
```diff
@@ -1,14 +1,13 @@
...
-  "name": "HangHub",
-  "version": "1.0.1",
-  "description": "HangHub is a productivity tool that allows you to see users who are working on the same GitHub issue as you. It is available as a browser extension for Chrome and Firefox.",
-  "main": "src/index.js",
-  "scripts": {
-    "build": "webpack --mode=production",
-    "lint": "eslint ./src"
-  },
+  "name": "hang-hub-frontend",
+  "version": "0.0.1",
```
Comment (elszczepano) [inline frontend/package.json:3]: Why `0.0.1`?

File: frontend/package.json (hunk 3 above)
Comment (pjasiun) [reply frontend/package.json]: Ah, I see you moved it. Still, maybe we should provide some, simple description for backend and the frontend part of this plugin. Explain that the frontend part if the browser extension for Chrome and FF and the backend part support the communication between clients and do not store any users data permanently.

File: frontend/src/renderCollaboratorList.jsx [hunk 7]
```diff
@@ -6,40 +6,40 @@
...
+		</div>
+	);
+}
 
-	function getPageType( pageType ) {
-		return pageType === 'issues' ? 'issue' : 'pull request';
-	}
+function Collaborator( { user } ) {
+	return (
+		<span key={ user.id } className="css-truncate js-issue-assignees sidebar-assignee">
```
Comment (elszczepano) [inline frontend/src/renderCollaboratorList.jsx]: If we used `key` property in line **19** do we still need it there? IMO the key property there is redundant.

File: frontend/package.json (hunk 3 above)
Comment (ma2ciek) [reply frontend/package.json]: Do we need this description everywhere? We don't' publish this package on NPM so this description won't be used anywhere.

File: package.json [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+  "version": "0.0.1",
```
Comment (elszczepano) [inline package.json:3]: The same question as before: Why `0.0.1`? If we want to set version `0.0.1` we should also update tags with release versions.
Also, it can be problematic whit publishing it on Chrome Web Store and Firefox Add-ons - After every code change, we need to update the version. Currently, there is a `1.0.1` version.
Another question: why you didn't change a version in `backend/package.json`?

File: .eslintrc.js (hunk 5 above)
Comment (ma2ciek) [reply .eslintrc.js:24]: We use Preact, not React. So we actually can't use React's newest features.

File: .eslintrc.js (hunk 1 above)
Comment (ma2ciek) [reply .eslintrc.js]: I don't know if they are compatible with the Preact. I'll check it. Actually, I'm an opponent of creating runtime type validation if it can be omitted.

File: frontend/package.json (hunk 6 above)
Comment (ma2ciek) [reply frontend/package.json:3]: We don't publish this file. I don't see any reason to update this file later.

File: frontend/src/renderCollaboratorList.jsx (hunk 7 above)
Comment (ma2ciek) [reply frontend/src/renderCollaboratorList.jsx]: Actually, it should be used on line 19 only. This was a bug before.

File: package.json (hunk 8 above)
Comment (ma2ciek) [reply package.json:3]: We don't need to update this version at any time. And the version field is required for package.json file.

Comment (ma2ciek) [reply package.json:3]: Why you edited my answer? ;p

Comment (elszczepano) [reply package.json:3]: > And the version field is required for package.json file.

*`manifest.json`. Yes, my bad.

Comment (elszczepano) [reply package.json:3]: @ma2ciek by mistake, sorry. I returned the previous version :P

File: .eslintrc.js (hunk 1 above)
Comment (ma2ciek) [reply .eslintrc.js]: And with propTypes, we'd need to introduce the development build as well. propTypes are cut out from the production build.

**Comment by ma2ciek** [comment]
> Another question: why you didn't change a version in backend/package.json?

I'll change it to 0.0.1 too. I also don't see any reason to specify it. Because of the same reason as above. We won't publish this package and we won't create any separate changelog for that package. The backend just needs to be in sync with the frontend. Nobody will use backend as a dependency.

File: .eslintrc.js (hunk 1 above)
Comment (ma2ciek) [reply .eslintrc.js]: I checked it. `propTypes` works, but we'd need to build code in the development mode and intensively check console for the warnings since we don't have tests for the package. So I'm against it.
